programmers-coding-test/soongu/dfs-bfs/미로탐색.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const [n, m] = [5, 6];
+
+const graph = [
+                [1, 0, 1, 0, 1, 0],
+                [1, 1, 1, 1, 1, 1],
+                [0, 1, 0, 0, 0, 0],
+                [1, 1, 0, 1, 1, 1],
+                [1, 1, 1, 1, 0, 1]
+            ];
+
+const dx = [-1, 1, 0, 0];
+const dy = [0, 0, -1, 1];
+
+function bfs(x, y) {
+    const queue = [[x, y]];
+    
+    while(queue.length !== 0) {
+        [x, y] = queue.shift();
+
+        for (let i = 0; i < 4; i++) {
+            let nx = x + dx[i];
+            let ny = y + dy[i];
+
+            if (nx < 0 || nx >= n || ny < 0 || ny >= m) {
```

Comment:
문제 요구사항은
사람 시작 위치 : 1, 1
도착 위치 : n, m
이므로 아래처럼 조건식을 수정해주는게 맞습니다
```suggestion
            if (nx < 0 || nx > n || ny < 0 || ny > m) {
```

Reply:
if (nx < 0 || nx > n || ny < 0 || ny > m) 방식으로 하면 n에서 undfined걸려요 ㅠ